src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerArgumentBinder.cs

```diff
@@ -65,57 +66,56 @@ public class ControllerArgumentBinder : IControllerActionArgumentBinder
             if (actionDescriptor.BoundProperties.Count == 0 &&
                 actionDescriptor.Parameters.Count == 0)
             {
-                return Task.FromResult<IDictionary<string, object>>(
-                    new Dictionary<string, object>(StringComparer.Ordinal));
-            }
-            else if (actionDescriptor.BoundProperties.Count == 0)
-            {
-                return BindActionArgumentsCoreAsync(controllerContext, actionDescriptor);
-            }
-            else
-            {
-                return BindActionArgumentsAndPropertiesCoreAsync(
-                    controllerContext,
-                    controller,
-                    actionDescriptor);
+                return TaskCache.CompletedTask;
             }
-        }
 
-        private async Task<IDictionary<string, object>> BindActionArgumentsCoreAsync(
-            ControllerContext controllerContext,
-            ControllerActionDescriptor actionDescriptor)
-        {
-            var valueProvider = await CompositeValueProvider.CreateAsync(
-                controllerContext,
-                controllerContext.ValueProviderFactories);
-
-            var actionArguments = await PopulateArgumentsAsync(
-                controllerContext,
-                actionDescriptor.Parameters,
-                valueProvider);
-            return actionArguments;
+            return BindArgumentsCoreAsync(controllerContext, controller, arguments);
```

Comment:
Get the comments elsewhere about the `public` API changes. All fine on the 3 overlapping things in this PR.